src/ui/src/pages/app/monitors/Index/components/MonitorForm/CustomConfiguration.tsx

```diff
@@ -1,37 +1,121 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { EuiFormRow, EuiFieldText, EuiTextArea } from '@elastic/eui';
+import {
+  EuiFormRow,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiComboBox,
+  EuiButton,
+  EuiTextArea,
+  EuiSpacer,
+} from '@elastic/eui';
 
-const CustomConfiguration: React.FC<{ setConfiguration: any }> = ({
-  setConfiguration,
-}) => {
-  const [sql, setSql] = useState('');
-  const [thresholds, setThresholds] = useState('');
+const ThresholdForm: React.FC<any> = ({threshold, setThreshold}) => {
+  const [selectedOptions, setSelectedOptions] = useState([]);
+  const [value, setValue] = useState('');
+
+  useEffect(() => {
+    const retrieveOperator = (selectedOptions:any) => {
+      if (selectedOptions.length > 0) {
+        return selectedOptions[0].value;
+      } else {
+        return null
+      };
+    }
+    setThreshold({
+        id: threshold.id,
+        operator: retrieveOperator(selectedOptions),
+        value: value,
+    }, threshold.id);
+  }, [value, selectedOptions, threshold])
+
+  return (
+    <EuiFlexGroup>
+      <EuiFlexItem>
+      <EuiFormRow label="Operator">
+      <EuiComboBox
+          placeholder="Select a single option"
+          singleSelection={{ asPlainText: true }}
+          options={
+            ['eq', 'ne', 'gt', 'ge', 'lt', 'le'].map((el: any) => {
+              return {
+                label: el,
+                value: el,
+              }
+            })
+          }
+          selectedOptions={selectedOptions}
+          onChange={(e: any) => setSelectedOptions(e)}
+        />
+      </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem>
+      <EuiFormRow label="Value (Numeric)">
+        <EuiFieldText
+          value={value}
+          onChange={(e: any) => setValue(e.target.value)}
+        />
+      </EuiFormRow>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+}
 
+const CustomConfiguration: React.FC<{setConfiguration: any}> = ({setConfiguration}) => {
+  const [sql, setSql] = useState('');
+  const [thresholds, setThresholds] = useState<any[]>([]);
+  
   function updateFormState(setState: any) {
     setState();
-    setConfiguration({ sql: sql, thresholds: [] });
+    setConfiguration({ sql: sql, thresholds: thresholds.map((el:any) => {return {operator: el.operator, value: parseFloat(el.value)}}) });
   }
 
+  const handleAddThreshold = () => {
+        const num_thresholds = thresholds.length;
```

Comment:
Minor callout - this is fine since we don't handle deletion at the moment. If we did, setting it to length would cause repeat values

Reply:
ooo good callout